d2renderers/d2svg/d2svg.go

```diff
@@ -1010,6 +1015,34 @@ func Render(diagram *d2target.Diagram) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
+// sortObjects sorts all diagrams objects (shapes and connections) in the desired drawing order
+// the sorting criteria is:
+// 1. zIndex, lower comes first
+// 2. two shapes with the same zIndex are sorted by their level (container nesting), containers come first
+// 3. two shapes with the same zIndex and same level, are sorted in the order they were exported
+// 4. shape and edge, shapes come first
+func sortObjects(allObjects []d2target.DiagramObject) {
+	sort.SliceStable(allObjects, func(i, j int) bool {
+		// first sort by zIndex
+		iZIndex := allObjects[i].GetZIndex()
+		jZIndex := allObjects[j].GetZIndex()
+		if iZIndex != jZIndex {
+			return iZIndex < jZIndex
+		}
+
+		// then, if both are shapes, the containers come first
+		iShape, iIsShape := allObjects[i].(d2target.Shape)
+		jShape, jIsShape := allObjects[j].(d2target.Shape)
+		if iIsShape && jIsShape {
+			return iShape.Level < jShape.Level
+		}
+
+		// then, shapes come before connections
+		_, jIsConnection := allObjects[j].(d2target.Connection)
+		return iIsShape && jIsConnection
```

Comment:
u technically don't need this, cuz u add objects before connections already and it's sortstable. but i guess it's good to clarify

Reply:
you say this is always true, like that it can't happen for _i to be a connection and j be a shape_?

Reply:
no i mean, if u remove these 3 lines and return 0 or something (tie), then sortstable would already have the right order (cuz it's stable)